sdk/scheduler/src/test/resources/ranges.yml

```diff
@@ -0,0 +1,24 @@
+name: "ranges"
+pods:
+  meta-data:
+    count: 2
+    tasks:
+      some-task:
+        goal: RUNNING
+        cmd: "some command"
+        ports:
+          name1:
+            port: 0
+            env-key: key1
+            ranges:
+              - begin: 1
+                end: 21
+              - begin: 2000
+                end: 5050
+          name2:
+            port: 0
+            ranges:
+              - begin:
```

Comment:
Assuming the following is incorrect:
```
...
...
  - begin:
    end: 21
  - begin: 
    end: 42
  - begin: 44
    end: 
```
where is this validated ? Is it possible to add a test to test that this is an invalid configuration?

Reply:
this defines an open interval (left or right) this means give me any port less than 21 or any port less than 42 or any port greater than 44, this yml is validated here validPortRangesTest()

Reply:
Oh, I am torn here. Although it makes sense when you explained it, i am inclined to make the check more stringent so that the above would be a invalid config. The user can rephrase this as 
```
  - begin:
    end: 42
  - begin: 44
    end: 
```
(which would be exactly the same thing - right?!)
I felt it may be confusing to keep empty values for `begin: ` or `end: ` when they are not the first (or last) occurrences in `ranges: ` but I don't have a strong preference for either.

Reply:
Also, perhaps add a config validator here - https://github.com/mesosphere/dcos-commons/tree/master/sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate (for example the user cannot decrease the port range with service update/upgrade option)?!

Reply:
ya that would be the same thing, I just copy pasted the yml from a dummy sample I had which is where the overlap comes from, ya I agree it can be confusing, but @vishnu2kmohan may have thoughts

Reply:
I dont think we should restrict users from updating the port range constraints ill follow up this PR with corresponding changes to the yml reference and developer guide

Reply:
closing this for now, we can make changes based on feedback from the yml changes but for now I think its good to be as accepting as possible